### DISCUSSION TITLE
How to use custom object detection tflite model in java code

### DISCUSSION THREAD

**Discussion by hadronknights** [FIRST-Tech-Challenge/FtcRobotController#249] [General] [answered]
The team has created custom object detection model and copied the tflite file under TeamCode/assets. We see the file is loaded by asset manager but fails and rc app just dies when invoking tfod.loadModelFromAsset(TFOD_MODEL_ASSET, LABELS)

Any help to detail out the steps to use the custom model in TFOD.

Thank you

**Comment by texasdiaz** [comment, marked as answer]
What timing! I recommend checking out the ftc-ml user manual, section 6. It contains instructions on how to use custom object models.

https://storage.googleapis.com/ftc-ml-firstinspires-prod/docs/ftc-ml_manual_2021.pdf

**Comment by hadronknights** [comment]
Ty for sharing ftc-ml tooling. The team was able to figure out why the load was not working. It was really issue with tflite model maker creating tensorflow 1 models.

Btw, the buttons like upload video  etc  in ftc-ml shows disabled and greyed out. Is the ml tooling available to the teams?

**Reply by cmacfarl** [reply]
Yes, the ml tooling is available for all teams.

However, for Youth Protection Policy reasons only users with the FIRST dashboard role of Lead Mentor/Coach1 or Lead Mentor/Coach2 are allowed to upload video.

If you know who has those roles on your team, you can ask them to upload video for you, and then all other features are available to team members per the instructions in the manual for granting team members access to the system.

**Comment by hadronknights** [comment]
We need desperate help with FTC ML tool. We had 167 minutes left of training time before our last training attempt. We started the training but after 25 minutes the training failed and now it say remaining training time as 0. Can someone please help with resetting the clock for team 12600. We have new team shipping element and trying to get the custom model created for the upcoming league championship on coming Saturday. Appreciate any help. Thank you

**Reply by texasdiaz** [reply]
This is not the forum for questions about FTC-ML. Please refer to the FTC-ML community forum for that.

https://community.ftclive.org/
